### DISCUSSION TITLE
Settings fails to load b/c of bad JSON caused by kernel update.

### DISCUSSION THREAD

**Discussion by elliotn** [rern/rAudio#1164] [Bug Reports]
I ran an upgrade to os via pacman and system menu option failed due to bad json:

, "soundprofileconf" : [

,60
,1500
,1000
]

Notice the leading comma before 60. That's the problem. There's a missing value.

Apparently, kernel parameters changed location in the filesystem (https://forum.endeavouros.com/t/sysctl-output-changed-from-kernel-5-10-to-5-13-why/17097).

This will no longer work, and produces file not found:
  $ sysctl kernel.sched_latency_ns

This will provide the value:
  $ cat /sys/kernel/debug/sched/latency_ns

When I updated /srv/http/bash/system-data.sh
from  $( sysctl kernel.sched_latency_ns | awk '{print $NF}' | tr -d '\0' )
to  $( cat /sys/kernel/debug/sched/latency_ns | awk '{print $NF}' | tr -d '\0' )
settings now displays!

It looks like "sysctl kernel.sched_latency_ns" is in system.sh & system-data.sh and probably needs to be updated there. (I'm too tired to bang any more on this tonight, but Settings now works for me!)

Thx.

**Comment by clriis** [comment]
I seem to be having the same issue. Played around with player items and now Settings have disappeared.
I'm using the latest update
I'm a novice in fiddling with Arch system so please bear with me, but:

I do get the value: 18000000 when `cat /sys/kernel/debug/sched/latency_ns`

I don't have `/srv/http/bash/system-data.sh`
But I got `/srv/http/bash/settings/system-data.sh`
This later file does not contain anything with *latency_ns*

So I'm kind of stuck!

**Reply by elliotn** [reply]
I'm not sure what is causing your settings to not show up, but here's how I found out what was wrong with my settings JSON:
- Copy/paste JSON from browser into a file (xxx.json)
- Run pretty print to clean up the JSON:  json_pp < xxx.json

This is what my output looked like:

malformed JSON string, neither array, object, number, string or atom, at character offset 1579 (before ",60\n,1500\n,1000\n]...") at /usr/bin/json_pp5.30 line 52.

From there I backtracked in the settings scripts. 

Good luck!

**Comment by cmh714** [comment]
Often even if the main player ui does not render you can still get to the other pages....
http://YOUR_IP_HERE/settings.php?p=system
